core/core-spacemacs-buffer.el

```diff
@@ -356,9 +374,13 @@ MIN-WIDTH is the minimal width of the frame, frame included.  The frame will not
     (if (not (file-exists-p content))
         (insert content)
       (insert-file-contents content)
-      (goto-char (point-max))
-      (when (eq ?\n (char-before))    ;; remove additional newline at eof
-        (delete-char -1)))
+      (if spacemacs-buffer-note-preview-lines
+          (progn (spacemacs-buffer//note-remove-after-line
```

Comment:
first form of progn can be moved out of if form.
Because it has no effect when spacemaca-notes-preview-lines is nil.

Reply:
Moving the function: `spacemacs-buffer//note-remove-after-line` 
outside the condition: `(if spacemacs-buffer-note-preview-lines`
could be read as if something from the note is being removed by default.

Therefore I renamed the function to make it clear that it only removes something if a note preview is requested.
`spacemacs-buffer//if-note-preview-remove-rest-of-note`

And removed the if statement, by adding two newlines to the end of the function
`spacemacs-buffer//if-note-preview-remove-rest-of-note`